### PR TITLE
Update compatibility for build-tools-2.2.0-alpha6

### PR DESCRIPTION
**Pull request by Naitbit**
-Bumps build tools version so it works with currently newest alpha of build tools (hopefully it will work with final 2.2.0)
-`globalScope` property for some reason returned `TransformGlobalScope` instead of `GlobalScope`. Calling `getGlobalScope()` seems to return value that we want.
-Please verify whether I've updated `FullCompilationSpec` correctly.

### CODE REVIEW

**Comment by pieces029** [comment]
Thanks for the pull request. I don't want to merge this into the mainline since this would make the plugin depend on alpha software. It's good to know that it does currently work but will want to hold off for now. 

If you could add a comment to the `getGlobalScope` as to why the get is being explicitly used that would be great, and the only thing missing from the `FullCompilationSpec` is the compatibility test for the library plugin.

**Comment by Naitbit** [comment]
Sure, I've expected that there will be no release with alpha dependency. I've just wanted to make sure that plugin will not stop working with final 2.2 build tools.
I've added comment and test as requested. I hope that it will be helpful to someone :)

**Comment by pieces029** [comment]
Thanks! Yeah, as soon as the latest tools get pushed I'll merge this and drop the `-apha` or let you do the honors by updating this PR.

**Comment by uKL** [comment]
Is there any temporary workaround until release?

**Comment by Naitbit** [comment]
You can:
-download my branch, build it and copy resulting .jar into you local maven repo and use  `1.0.1-SNAPSHOT` as classpath
-or downgrade build tools to `2.1.2`

**Comment by uKL** [comment]
I did the first one and added classpath dependency to a jar file, within the project. Thanks!

**Comment by pieces029** [comment]
You should also be able to pull this branch and do `./gradlew publishMavenLocal` then in your project add `mavenLocal()` as a repository. When you reference the snapshot you will actually be using the published branch. You may want to change the version number to something like `1.0.1-2.2.0-alpha` just to ensure you are grabbing the correct jar file.

**Comment by uKL** [comment]
It's easier to rely on a JAR dependency (I don't have to compile it each time on my CI). But thanks!

**Comment by ubuntudroid** [comment]
Is this to be merged and incorporated into a release anytime soon? Just stumbled over this issue while setting up spock for our Android project.

**Comment by Naitbit** [comment]
AFAIK it will be merged once build tools/AS 2.2 will be released. They are currently at RC so it will probably be soon.

**Comment by ubuntudroid** [comment]
Awesome, that's great to hear! :) Then we'll probably postpone our
experiments until then.

File: groovy-android-gradle-plugin/src/test/groovy/groovyx/functional/FullCompilationSpec.groovy [hunk 1]
```diff
@@ -27,7 +27,7 @@ import static groovyx.internal.TestProperties.compileSdkVersion
  * Complete test suite to ensure the plugin works with the different versions of android gradle plugin.
  * This will only be run if the system property of 'allTests' is set to true
  */
-@IgnoreIf({ !allTests })
+@IgnoreIf({ false })
```
Comment (pieces029) [inline groovy-android-gradle-plugin/src/test/groovy/groovyx/functional/FullCompilationSpec.groovy]: Why was this changed?

**Comment by pieces029** [comment]
Can you bump to 2.2.0-rc2 and rebase so I can put this into master. The people can try it out as a snapshot.

File: groovy-android-gradle-plugin/src/test/groovy/groovyx/functional/FullCompilationSpec.groovy (hunk 1 above)
Comment (Naitbit) [inline groovy-android-gradle-plugin/src/test/groovy/groovyx/functional/FullCompilationSpec.groovy]: No good reason, it was accident. I'll revert that. I wanted to make sure that this tests run locally, but unfortunately I did up committing that.

**Comment by Naitbit** [comment]
@pieces029 I did bump the build tools version, but now I have issue with
`should compile android app with`* tests failing when the gradle version is different than `2.14.1`. Adding `System.properties['com.android.build.gradle.overrideVersionCheck'] = 'true'` makes them pass but I'm not sure whether it is good solution. Please take a look, you may find a better one.

**Comment by pieces029** [comment]
Which test is failing? I've seen this before where the Android Team changes external dependencies so older tests need to be changed a little bit.

**Comment by Naitbit** [comment]
```
Failed tests

    FullCompilationSpec. should compile android app with java:1.6, android plugin:1.1.0, gradle version: 2.10
    FullCompilationSpec. should compile android app with java:1.6, android plugin:1.3.0, gradle version: 2.10
    FullCompilationSpec. should compile android app with java:1.6, android plugin:1.5.0, gradle version: 2.10
    FullCompilationSpec. should compile android app with java:1.7, android plugin:1.5.0, gradle version: 2.11
    FullCompilationSpec. should compile android app with java:1.7, android plugin:1.5.0, gradle version: 2.12
    FullCompilationSpec. should compile android app with java:1.7, android plugin:2.0.0, gradle version: 2.13
    FullCompilationSpec. should compile android app with java:1.7, android plugin:2.1.2, gradle version: 2.14
    FullCompilationSpec. should compile android library with java:1.6 and android plugin:1.1.0, gradle version:2.10
    FullCompilationSpec. should compile android library with java:1.6 and android plugin:1.3.0, gradle version:2.10
    FullCompilationSpec. should compile android library with java:1.6 and android plugin:1.5.0, gradle version:2.10
    FullCompilationSpec. should compile android library with java:1.7 and android plugin:1.5.0, gradle version:2.10
    FullCompilationSpec. should compile android library with java:1.7 and android plugin:1.5.0, gradle version:2.11
    FullCompilationSpec. should compile android library with java:1.7 and android plugin:1.5.0, gradle version:2.12
    FullCompilationSpec. should compile android library with java:1.7 and android plugin:2.0.0, gradle version:2.13
    FullCompilationSpec. should compile android library with java:1.7 and android plugin:2.1.2, gradle version:2.14

```

Reason:
```
Expected no exception to be thrown, but got 'org.gradle.testkit.runner.UnexpectedBuildFailure'
	at spock.lang.Specification.noExceptionThrown(Specification.java:118)
	at groovyx.functional.FullCompilationSpec.should compile android app with java:#javaVersion, android plugin:#androidPluginVersion, gradle version: #gradleVersion(FullCompilationSpec.groovy:225)
Caused by: org.gradle.testkit.runner.UnexpectedBuildFailure: Unexpected build execution failure in /tmp/junit1549353321413592455 with arguments [assemble, test]

Output:

FAILURE: Build failed with an exception.

* Where:
Build file '/tmp/junit1549353321413592455/build.gradle' line: 13

* What went wrong:
A problem occurred evaluating root project 'test-app'.
> Failed to apply plugin [id 'com.android.application']
   > Minimum supported Gradle version is 2.14.1. Current version is 2.10. If using the gradle wrapper, try editing the distributionUrl in /home/sebastian/projects/nait-clone/groovy-android-gradle-plugin/groovy-android-gradle-plugin/gradle/wrapper/gradle-wrapper.properties to gradle-2.14.1-all.zip

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 1.669 secs

	at org.gradle.testkit.runner.internal.DefaultGradleRunner$1.execute(DefaultGradleRunner.java:222)
	at org.gradle.testkit.runner.internal.DefaultGradleRunner$1.execute(DefaultGradleRunner.java:219)
	at org.gradle.testkit.runner.internal.DefaultGradleRunner.run(DefaultGradleRunner.java:282)
	at org.gradle.testkit.runner.internal.DefaultGradleRunner.build(DefaultGradleRunner.java:219)
	at groovyx.functional.FunctionalSpec.runWithVersion(FunctionalSpec.groovy:45)
	at groovyx.functional.FullCompilationSpec.should compile android app with java:#javaVersion, android plugin:#androidPluginVersion, gradle version: #gradleVersion(FullCompilationSpec.groovy:222)
```

**Comment by pieces029** [comment]
Whoa, so it looks like they all need 2.14.1?

**Comment by pieces029** [comment]
Could you squash your commits and I'll get this merged in? I take a look at the weird gradle compatibility stuff later.

**Comment by Naitbit** [comment]
Sure, I've squashed the commits. Thanks for taking a look on compatibility.

**Comment by pieces029** [comment]
Cool, Thanks!
